encoding/front_matter.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+// Copyright (c) Jason Campbell. MIT license.
+// https://github.com/jxson/front-matter/blob/36f139ef797bd9e5196a9ede03ef481d7fbca18e/index.js
+
+import { parse } from "./yaml.ts";
+
+const pattern = "^(" +
+  "\\ufeff?" + // Maybe byte order mark
+  "(= yaml =|---)" +
+  "$([\\s\\S]*?)" +
+  "^(?:\\2|\\.\\.\\.)\\s*" +
+  "$" +
+  (Deno.build.os === "windows" ? "\\r?" : "") +
+  "(?:\\n)?)";
+const regex = new RegExp(pattern, "m");
+
+export type Extract<T> = {
+  frontMatter: string;
+  body: string;
+  attrs: T;
+};
+
+/**
+ * Extracts front matter from a string.
+ * @param str String to extract from.
+ *
+ * ```ts
+ * import { extract } from "https://deno.land/std@$STD_VERSION/encoding/front_matter.ts";
+ * import { assertEquals } from "https://deno.land/std@$STD_VERSION/testing/asserts.ts";
+ *
+ * const { attrs, body, frontMatter } = extract<{ title: string }>("---\ntitle: Three dashes marks the spot\n---\n");
+ * assertEquals(attrs.title, "Three dashes marks the spot");
+ * assertEquals(body, "");
+ * assertEquals(frontMatter, "---\ntitle: Three dashes marks the spot\n---\n");
```

Comment:
Hello, the test here should be
`assertEquals(frontMatter, "title: Three dashes marks the spot");`
without the front matter payload.

Reply:
Good catch. Thanks!